apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/Trace/CloudTrace.cs

```diff
@@ -119,16 +117,11 @@ public static void Initialize(string projectId, HttpApplication application, Tra
         private void BeginRequest(object sender, EventArgs e)
         {
             TraceHeaderContext headerContext = TraceHeaderContextUtils.CreateContext(HttpContext.Current.Request);
-            TraceOptions headerOptions = _headerFactory.CreateOptions(headerContext);
-
             // If the trace header says to trace or if the rate limiter allows tracing continue.
-            if (!headerOptions.ShouldTrace)
+            if (headerContext.ShouldTrace == false || (headerContext.ShouldTrace == null &&
+                !_rateFactory.CreateOptions().ShouldTrace))
```

Comment:
Just to check, CreateOptions is what updates the count in the rate limiter, isn't it? That's still not obvious from the naming - possibly something to revisit at some point.